internal/controller/bootstrap/controlplane_bootstrap_controller.go

```diff
@@ -155,6 +156,25 @@ func (c *ControlPlaneController) Reconcile(ctx context.Context, req ctrl.Request
 	)
 
 	if config.Spec.K0s != nil {
```

Comment:
A few thoughts:
- Should we set the field even if user didn't provide us any custom k0s config?
- At the end of this if statement, we call `kutil.ReconcileDynamicConfig`, but we probably need to remove the etcd node name from the DynamicConfig. 
- Should the most of the code of this statement be in the control plane controller instead of bootstrap controller?

Reply:
> * Should we set the field even if user didn't provide us any custom k0s config?

Agree. It is already used etcd node name as default value if kine is not set so I think is safe assume that. Added
> * At the end of this if statement, we call `kutil.ReconcileDynamicConfig`, but we probably need to remove the etcd node name from the DynamicConfig.

Im not sure if I follow your point. It is not removed `spec.storage` in `kutil.ReconcileDynamicConfig` from the DynamicConfig?
> * Should the most of the code of this statement be in the control plane controller instead of bootstrap controller?

If you mean to move most of the logic to the place when we create `bootstrapv1.K0sControllerConfig` Im ok with that. But I think that change should be done carefully, Im not sure exactly what it might imply rn. Maybe addressing that refactoring in another PR makes more sense, wdyt?

Reply:
> Im not sure if I follow your point. It is not removed spec.storage in kutil.ReconcileDynamicConfig from the DynamicConfig?

Sorry, missed it. Yeah, that's good then. 

> Maybe addressing that refactoring in another PR makes more sense, wdyt?

Sorry, I was not clear enough: of course, it's better to do it in a separate PR.

Reply:
https://github.com/k0sproject/k0smotron/issues/798